recorder/meter.go

```diff
@@ -28,9 +30,14 @@ func NewMeter(c MeterConfig) otelmetric.Meter {
 
 	var err error
 
-	var exp *prometheus.Exporter
+	var opt []exporter.Option
+	if c.Reg != nil {
+		opt = append(opt, exporter.WithRegisterer(c.Reg))
+	}
```

Comment:
If the parameter is provided, we inject it as option for the underlying exporter.